enclave-runtime/src/rpc/nostr_utils.rs

```diff
@@ -0,0 +1,97 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+use crate::{String, Vec};
+use itp_time_utils::{duration_now, now_as_secs, Duration};
+use log::info;
+use nostr::{prelude::*, types::time::TimeSupplier, Event, Timestamp};
+use tungstenite_sgx as tungstenite;
+
+use nostr::ClientMessage;
+
+use tungstenite::Message as WsMessage;
+
+pub struct DemoTimeProvider;
+
+/// Time related operations are rather limited in `no_std` environments, which is why the `nostr`
+/// crate provides a trait that can help `no_std` users have time related operations.
+impl TimeSupplier for DemoTimeProvider {
```

Comment:
I think this should also be done for the trait name to be consistent, but does this incur a lot of overhead because it is in an external crate?